minvws/nl-kat-coordination
# Containerize all non-docker boefjes

### Changes

As the title suggests, this PR introduces a new OCI image having the requirements of all non-docker boefjes. Through the `oci_image` and `oci_arguments` we now specify the right script within this generic image. This, in short, is going to be the biggest jump in containerizing our boefjes, converting _a total of 27 boefjes_. In follow-up PRs, the 13 docker-boefjes will be turned into specialized images having the same interface as well.

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3698

### Demo

-

### QA notes

This is going to be a chunk of work, but to really verify these changes work we need to test whether the following boefjes (by their directory name because I could copy that easily) still run and produce the output we expect:

- kat_dns
- kat_snyk
- kat_binaryedge
- kat_censys
- kat_crt_sh
- kat_cve_2023_34039
- kat_cve_2023_35078
- kat_dicom
- kat_dns_version
- kat_dns_zone
- kat_external_db
- kat_fierce
- kat_green_hosting
- kat_leakix
- kat_rdns
- kat_rpki
- kat_security_txt_downloader
- kat_service_banner
- kat_shodan
- kat_webpage_analysis
- pdio_subfinder
- kat_adr_finding_types
- kat_cve_finding_types
- kat_cwe_finding_types
- kat_kat_finding_types
- kat_retirejs_finding_types
- kat_snyk_finding_types

Another interesting note would be to see if there are any significant performance degradations.

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
